zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -131,6 +129,6 @@ public void process(WatchedEvent event) {
 
   @Override
   public Set<String> getDomains(String clientUri) {
-    return clientUriToDomainNames.getOrDefault(clientUri, Collections.emptySet());
+    return clientUriToDomainNames.getOrDefault(clientUri, Sets.newHashSet());
```

Comment:
Ideally, this function should return an unmodifiable set. If you want to modify sets, the caller should prepare a modifiable set and add this set to it.